cpp/ql/src/Security/CWE/CWE-313/CleartextSqliteDatabase.ql

```diff
@@ -34,25 +28,51 @@ predicate sqlite_encryption_used() {
   any(FunctionCall fc).getTarget().getName().matches("sqlite%\\_key\\_%")
 }
 
-class Configuration extends TaintTrackingConfiguration {
-  override predicate isSource(Expr source) {
-    super.isSource(source) and source instanceof SensitiveExpr
+/**
+ *  Gets a field of the class `c`, or of another class contained in `c`.
+ */
+Field getRecField(Class c) {
+  result = c.getAField() or
+  result = getRecField(c.getAField().getUnspecifiedType().stripType())
```

Comment:
- Does this recursive field digging actually work? I don't see it changing any query results in our tests.

Reply:
The second case is needed for a couple of test cases, though that only uses one recursive step so I think this is a fair criticism.  I've added another test case (to the internal test) with two recursive steps.